lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -302,16 +312,41 @@ def write(chunk)
             entry['metadata']['severity'] = 'DEFAULT'
           end
 
+          if @service_name == CONTAINER_SERVICE
+            # TODO(a-robinson): Consider falling back to parsing the namespace,
+            # pod, and container names out of the log name if we don't have this
+            # information, which depends on the kubernetes_metadata_filter
+            # filter plugin.
+            labels = {}
+            field_to_label(record['kubernetes'], 'namespace_id', labels,
+                                  "#{CONTAINER_SERVICE}/namespace_id")
+            field_to_label(record['kubernetes'], 'namespace_name', labels,
+                                  "#{CONTAINER_SERVICE}/namespace_name")
+            field_to_label(record['kubernetes'], 'pod_id', labels,
```

Comment:
No, names aren't mutable, and I'm not a big fan of this either, but the Google cloud logging and monitoring services prefer indexing by IDs rather than names. The IDs can then be mapped to names at query time as appropriate.

The main benefit is that it allows for distinguishing between two different objects with the same name that existed at different times (i.e. create foo, delete foo, create foo).